website/notebookconvert.py

```diff
@@ -1,39 +1,57 @@
 import os
 import re
 
+
 def add_header_to_markdown(folder, md):
     name = md[:-3]
-    with open(os.path.join(folder, md), 'r+', encoding='utf-8') as f:
+    with open(os.path.join(folder, md), "r+", encoding="utf-8") as f:
         content = f.read()
         f.truncate(0)
-        content = re.sub(r'style=\"[\S ]*?\"', '', content)
-        content = re.sub(r'<style[\S \n.]*?</style>', '', content)
+        content = re.sub(r"style=\"[\S ]*?\"", "", content)
+        content = re.sub(r"<style[\S \n.]*?</style>", "", content)
         f.seek(0, 0)
         f.write("---\ntitle: {}\nhide_title: true\nstatus: stable\n---\n".format(name) + content)
         f.close()
 
-def convert_notebook_to_markdown(file_path, outputdir):
-    print("Converting {} into markdown \n".format(file_path))
-    convert_cmd = 'jupyter nbconvert --output-dir=\"{}\" --to markdown \"{}\"'.format(outputdir, file_path)
+
+def convert_notebook_to_markdown(folder, nb, outputdir):
+    file_path = os.path.join(folder, nb)
+    print(f"Converting {file_path} into markdown")
+
+    # If the notebook contains cell outputs such as figures, a folder containing cell output images is generated alongside the markdown file
+    # By default, both the folder and files contain the notebook name. But spaces in the notebook name create linking errors in the generated markdown
+    # Therefore, we first generate the markdown file, output folder, and output files with no spaces
```

Comment:
Love this! But for figures could we upload them to our public blob (https://mmlspark.blob.core.windows.net/graphics) and use the url directly? Just want to be consistent with the structure of /notebooks folder.

Reply:
I definitely don't have an issue with this; however, I think that keeping the inline images in the notebook is advantageous because:
- The notebook creator doesn't have to export the inline images and upload them to blob each time they either add images or edit images
- The notebook creator doesn't have to add an extraneous notebook cell that points to the image from blob
- When a user runs the notebook, the unintended effect will be this (user will see the graph twice -- once from generated code and once from markdown link):

![image](https://user-images.githubusercontent.com/64443771/140466207-78d72ede-b7e9-435e-8eda-84197be3925e.png)

What are your thoughts? :smile:

Reply:
@ms-kashyap, how do you get it so that its the only cell of output shown? do you manually delete all of the other output?

Reply:
If you want to clear output of all cells, then there's a button for that in VS Code
![image](https://user-images.githubusercontent.com/64443771/140555030-49a9a3fd-d938-4cd4-b464-a992a1519e79.png)

However, if you want to keep the output of one cell, then yeah you need to go to every other cell and clear its output manually and painstakingly 😅 
![image](https://user-images.githubusercontent.com/64443771/140555199-63517e1d-3dee-4411-99e7-fe45c03866be.png)

Reply:
For @memoryz and @mhamilton723 's context, @serena-ruan and I talked offline and we are going with pointing to the matplotlib figures via blob urls in markdown cells below the code (and clearing all outputs).

If we encounter a notebook where we HAVE to keep cell outputs (which would in turn generate the extra folder + files), then we can rethink this at that point.